cmd/cloud.go

```diff
@@ -313,9 +312,16 @@ func (c *cmdCloud) run(cmd *cobra.Command, args []string) error {
 		logger.WithField("run_status", testProgress.RunStatusText).Debug("Test finished")
 	}
 
+	//nolint:stylecheck,golint
 	if testProgress.ResultStatus == cloudapi.ResultStatusFailed {
+		// Although by looking at [ResultStatus] and [RunStatus] isn't self-explanatory,
+		// the scenario when the test run has finished, but it failed is an exceptional case for those situations
+		// when thresholds have been crossed (failed). So, we report this situation as such.
+		if testProgress.RunStatus == cloudapi.RunStatusFinished {
```

Comment:
If I got it correctly, we have `RunStatus`es specifically for those situations (see https://github.com/grafana/k6/blob/master/cloudapi/run_status.go#L16-L20).

Also, my assumption is that `RunStatusAbortedThreshold` doesn't apply here, because in this case the test hasn't been aborted (`abortOnFail: true`), just finished and failed at the end, because unsuccessful threshold checks.

And yes, that's what I got after a conversation with @d14c and @Griatch, so maybe they can confirm it.

Reply:
Additionally, I know there's other edge cases that we may want to handle within this piece of code (other of the aformentioned `RunStatus`es), but as I left as note in the PR description, I'd prefer to just focus on this edge case here, and leave the other scenarios for upcoming, separated PRs.

Reply:
I agree with @olegbespalov that sounds like a risky assumption. Furthermore, I have the feeling that all the time that we need to read this code, it will required a lot of cognitive load.

Could we ask the backend to add a dedicated RunStatus for it? Like: `RunStatusFinishedWithCrossThresholds`.

Reply:
@joanlopez There are three possible outcomes here:
- `ABORTED_BY_THRESHOLD` this is a `run_status` end status for thresholds that are set to abort the test when crossed. In this case the test will never reach `run_status` `FINISHED`. 
- `run_status` FINISHED can still be reached if the threshold was not set up to abort the test. In this case the outcome will only be reflected in the `result_status` which is handled completely separately from `run_status`. 
- `result_status` False/True will start out as True and will switch to False only on a failed threshold afaik. So it's for example possible to have a test that has a `run_status` ABORTED_SYSTEM but with `result_status` True because the threshold was never crossed. 

For the upcoming public API we are introducing a 'success' status that is a little easier to reason around. Inviting @olegbespalov on this to make sure I get it right; but it's basically a combination of run/result status that will fail both on a threshold-cross and on a test that fail to complete properly.

Reply:
So, if I got it right, it looks like my assumptions are confirmed again by @Griatch. 

What do you think guys? @olegbespalov @codebien 
Do you want to go like this? Or wait until we have the new API with clearer information?

If we prefer to wait, I can close this PR, and create an issue with some context, and marked as blocked, waiting for the new public API. As you prefer!

Thanks!

Reply:
After checking the backend code and finding a few test run examples to back it up, I can confirm that any cloud test with a failed threshold will end up in the `RunStatusAbortedThreshold` irrespective of the `abortOnFail` flag. Therefore, checking `testProgress.ResultStatus == cloudapi.RunStatusAbortedThreshold` is sufficient here.

Reply:
Thanks for all input! it sounds all clear and ledgit and it seems like we could clean the `TODO`

```
// TODO: use different exit codes for failed thresholds vs failed test (e.g. aborted by system/limit)
```

>If we prefer to wait, I can close this PR, and create an issue with some context, and marked as blocked, waiting for the new public API. As you prefer!

I don't know about ETA of new API, but if we could bring value earlier that sounds better.

Reply:
I guess my question is how likely it is:
- that the test has finished 
- fails
- and it isn't a threshold

I guess it is fine if we have some false possitives, but still.

Reply:
Isn't that what previous comment from @Griatch explains? Like: 

```

Reply:
ah, I guess I did skim over that particular line a bit too much.

Again this seems like a strange behaviour, but I guess it is what it is .